File: JULY-2025/17-06-1015/p3202.dart
import 'dart:math';

class Solution {
  int maximumLength(List<int> nums, int k) {
    int maxLength = 2;

    for (int targetMod = 0; targetMod < k; targetMod++) {
      List<int> remainderDP = List.filled(k, 0);

      for (int num in nums) {
        int numMod = num % k;
        int requiredMod = (targetMod - numMod + k) % k;

        remainderDP[numMod] = remainderDP[requiredMod] + 1;
      }

      for (int length in remainderDP) {
        maxLength = max(maxLength, length);
      }
    }

    return maxLength;
  }
}

File: JULY-2025/17-06-1015/p3202.java
class Solution {
    public int maximumLength(int[] nums, int k) {
        
        
     int n=nums.length;
        int[][] dp=new int[k][n];
        int max=1;
        for(int i=0;i<k;i++){
            for(int j=0;j<n;j++){
                dp[i][j]=1;
            }
        }
        for(int i=1;i<n;i++){
            for(int j=0;j<i;j++){
                    int mod=(nums[i]+nums[j])%k;

            
                    dp[mod][i]=Math.max(dp[mod][i],dp[mod][j]+1);
                    max=Math.max(max,dp[mod][i]);

            }
        }
        return max;

        
    }
}